compiler/src/iree/compiler/Dialect/Flow/Transforms/AnnotateDispatches.cpp

```diff
@@ -163,6 +163,27 @@ static std::string getOpNameWithoutDialectName(Operation *op) {
 }
 
 static std::string summarizeLinalgOp(linalg::LinalgOp op) {
+  // Check if the body only contains a yield.
+  bool hasOnlyYield = true;
+  op.walk([&](Operation *child) {
+    if (op == child || isa<linalg::YieldOp>(child))
+      return WalkResult::advance();
+    hasOnlyYield = false;
+    return WalkResult::interrupt();
+  });
+
+  // Check if the indexing maps are only permutations.
+  bool hasOnlyPermutation = true;
+  for (AffineMap map : op.getIndexingMapsArray()) {
+    if (!map.isPermutation()) {
+      hasOnlyPermutation = false;
+    }
+  }
+
+  if (hasOnlyYield && hasOnlyPermutation) {
+    return "transpose";
+  }
+
   auto opName = op->getName().getStringRef();
   if (!opName.consume_front("linalg."))
     return "";
```

Comment:
This changes the dispatch name from `generic_xxx` to `transpose`. The below loop ranges and types information is dropped in the PR. How about we set `opName` to `transpose`, so we still keep the information?

Reply:
yeah, just change the base name so we keep the goodies

Reply:
fixed